common/common-interactor/src/interactor_helpers.rs

```diff
@@ -491,25 +384,91 @@ pub trait InteractorHelpers {
         }
     }
 
-    fn extract_log_based_on_shard(&mut self, config: &ActionConfig) -> Option<String> {
-        match &config.expected_log {
-            Some(logs) if logs.len() == 1 => Some(logs[0].clone()),
-            Some(logs) if logs.len() > 1 => match config.shard {
-                SHARD_1 => Some(logs[0].clone()),
-                _ => Some(logs[1].clone()),
-            },
-            _ => None,
+    fn build_expected_deposit_log(
+        &mut self,
+        mut config: ActionConfig,
+        token: Option<EsdtTokenInfo>,
+    ) -> Vec<ExpectedLogs<'static>> {
+        if config.shard != SHARD_1 {
+            return vec![];
+        }
+
+        let mut logs = Vec::new();
+        let deposit_token = config.expected_deposit_token_log.clone().or(token);
+
+        match deposit_token {
+            Some(t) => {
+                let topic_value = t.token_id.into_managed_buffer().to_string();
+                logs.push(log!(DEPOSIT_EVENT, topics: [DEPOSIT_EVENT, topic_value]));
+            }
+            None => {
+                logs.push(log!(DEPOSIT_EVENT, topics: [SC_CALL_LOG]));
+            }
         }
+
+        if let Some(additional_logs) = config.additional_logs.take() {
+            logs.extend(additional_logs);
+        }
+
+        logs
     }
 
-    fn extract_log_error_based_on_shard(&mut self, config: &ActionConfig) -> Option<String> {
-        match &config.expected_log_error {
-            Some(error) => match config.shard {
-                SHARD_0 => Some(error[0].clone()),
-                _ => None,
-            },
-            _ => None,
+    fn build_sovereign_deposit_logs(
+        &mut self,
+        main_token: &EsdtTokenInfo,
+    ) -> Vec<ExpectedLogs<'static>> {
+        let main_token_id_topic = main_token
+            .token_id
+            .clone()
+            .into_managed_buffer()
+            .to_string();
+
+        let burn_log = match main_token.token_type {
+            EsdtTokenType::Fungible => {
+                log!(ESDT_LOCAL_BURN_FUNC_NAME, topics: [main_token_id_topic])
+            }
+            _ => log!(ESDT_NFT_BURN_FUNC_NAME, topics: [main_token_id_topic]),
+        };
+
+        vec![burn_log]
+    }
+
+    fn build_expected_execute_log(
```

Comment:
After the fixes will discuss about this function. I want to understand which are the problematic tests which require so many if's 😁